CMakeLists.txt

```diff
@@ -97,8 +97,8 @@ ament_target_dependencies(${PROJECT_NAME}
 
   # We don't export these dependencies because their cmake is broken
   ASSIMP
-  QHULL
 )
+ament_target_dependencies(${PROJECT_NAME} SYSTEM QHULL)
```

Comment:
Shouldn't all these dependencies be `SYSTEM`?

Reply:
This should also work, I updated the PR to make all dependencies `SYSTEM`

Reply:
We had problems with overlayed workspaces in ROS1/catkin multiple times because `SYSTEM` was used for packages that can be built in the same workspace and should overlay system versions. This can be problematic because `-isystem` will not just silence warnings, but also change the order of these includes to be behind all `-I` includes.

I'm not sure about the situation in ament/ros2 though.

Reply:
It's somewhat involved, but the problem comes together because:
- ROS include paths for install workspaces are the same for packages in the same workspace.
- `-isystem` includes always override `-I` includes if both are specified for the same folder

So it can happen that two dependent packages both include the path from their shared workspace, but one of them declares it via `-isystem`. In that case the path will only be considered after **all** `-I` includes. When working with overlayed workspaces, it's quite likely though that one of them contains a different version of the other package and you will get broken builds.

That's why I would always only declare dependencies that are not involved in ROS (and that will never be built in a ROS workspace) as `SYSTEM`

Reply:
Would it then make sense to split up the dependencies that are ROS from those that are and only declare the non-ROS ones as SYSTEM?

Reply:
That a very valid argument from Michael. Given that tricky situation, I agree with Tyler to only declare non-ROS packages as SYSTEM. Sorry for the noise.